algorithms/src/polycommit/sonic_pc/mod.rs

```diff
@@ -116,9 +116,10 @@ impl<E: PairingEngine> PolynomialCommitment<E::Fr, E::Fq> for SonicKZG10<E> {
         enforced_degree_bounds: Option<&[usize]>,
     ) -> Result<(Self::CommitterKey, Self::VerifierKey), PCError> {
         let trim_time = start_timer!(|| "Trimming public parameters");
-        let max_degree = pp.max_degree();
+        let mut max_degree = pp.max_degree();
         if supported_degree > max_degree {
-            return Err(PCError::TrimmingDegreeTooLarge);
+            pp.powers_of_beta_g.download_up_to(supported_degree).map_err(|_| PCError::TrimmingDegreeTooLarge)?;
+            max_degree = pp.max_degree();
```

Comment:
We never hit this line because if we're in the error case then the `?` returns.

Reply:
I don't think we should hit that line in case there's an error, only if we succeed in downloading the needed powers.